rosidl_generator_java/resource/idl.cpp.em

```diff
@@ -9,102 +9,93 @@
 @#  - package_name (string)
 @#  - interface_path (Path relative to the directory named after the package)
 @#  - content (IdlContent, list of elements, e.g. Messages or Services)
+@#  - output_dir (Path)
+@#  - template_basepath (Path)
+@#  - typesupport_impl (string, the typesupport identifier of the generated code)
 @#######################################################################
 @{
-include_directives = set()
+import os
 
-jni_includes = [
-    'jni.h',
-]
-include_directives.update(jni_includes)
-std_includes = [
-    'cassert',
-    'cstdint',
-    'string',
-]
-include_directives.update(std_includes)
-rosidl_includes = [
-    'rosidl_generator_c/message_type_support_struct.h',
-]
-include_directives.update(rosidl_includes)
-rcljava_includes = [
-    'rcljava_common/exceptions.h',
-    'rcljava_common/signatures.h',
-]
-include_directives.update(rcljava_includes)
-}@
-@[for include in jni_includes]@
-#include <@(include)>
-@[end for]@
-
-@[for include in std_includes]@
-#include <@(include)>
-@[end for]@
-
-@[for include in rosidl_includes]@
-#include "@(include)"
-@[end for]@
-
-@[for include in rcljava_includes]@
-#include "@(include)"
-@[end for]@
-
-// Ensure that a jlong is big enough to store raw pointers
-static_assert(sizeof(jlong) >= sizeof(std::intptr_t), "jlong must be able to store pointers");
+from rosidl_cmake import expand_template
+from rosidl_parser.definition import Action
+from rosidl_parser.definition import Message
+from rosidl_parser.definition import Service
 
-using rcljava_common::exceptions::rcljava_throw_exception;
+# jni_package_name = package_name.replace('_', '_1')
 
-@{
-jni_package_name = package_name.replace('_', '_1')
 }@
 @
 @#######################################################################
 @# Handle messages
 @#######################################################################
 @{
-from rosidl_parser.definition import Message
-}@
-@[for message in content.get_elements_of_type(Message)]@
-@{
-TEMPLATE(
-    'msg.cpp.em',
-    package_name=package_name,
-    jni_package_name=jni_package_name,
-    message=message,
-    include_directives=include_directives)
+data = {
+    'package_name': package_name,
+    # 'jni_package_name': jni_package_name,
```

Comment:
Presumably we no longer need the `jni_package_name`; assuming that is the case, just remove this?

Reply:
Yep, removed: 29d1eac